mesa/batchrunner.py

```diff
@@ -75,24 +75,23 @@ def batch_run(
         data_collection_period=data_collection_period,
     )
 
-    total_iterations = len(kwargs_list) * iterations
+    total_iterations = len(kwargs_list)
     run_counter = count()
 
     results: List[Dict[str, Any]] = []
 
     with tqdm(total_iterations, disable=not display_progress) as pbar:
         if number_processes == 1:
-            for iteration in range(iterations):
-                for kwargs in kwargs_list:
-                    _, rawdata = process_func(kwargs)
-                    run_id = next(run_counter)
-                    data = []
-                    for run_data in rawdata:
-                        out = {"RunId": run_id, "iteration": iteration - 1}
-                        out.update(run_data)
-                        data.append(out)
-                    results.extend(data)
-                    pbar.update()
+            for iteration, kwargs in enumerate(kwargs_list):
+                _, rawdata = process_func(kwargs)
+                run_id = next(run_counter)
+                data = []
+                for run_data in rawdata:
+                    out = {"RunId": run_id, "iteration": iteration - 1}
```

Comment:
Btw, this is another bug that already exists in the original code: since `iteration` starts from 0, `iteration - 1` starts with -1 instead, which is wrong. We can fix it in another PR.